src/main/interface/tramp_protocol.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+#include "common/utils.h"
+#include "interface/tramp_protocol.h"
+
+#define TRAMP_SYNC_START                0x0F
+#define TRAMP_SYNC_STOP                 0x00
+#define TRAMP_COMMAND_SET_FREQ          0x46 // "F"
+#define TRAMP_COMMAND_SET_POWER         0x50 // "P"
+#define TRAMP_COMMAND_ACTIVE_STATE      0x49 // "I"
+#define TRAMP_COMMAND_GET_CONFIG        0x76 // "v"
+#define TRAMP_FRAME_LENGTH              16
+#define TRAMP_HEADER_LENGTH             2
+
+typedef enum trampFrameType_e {
+    TRAMP_FRAME_SET_FREQUENCY,
+    TRAMP_FRAME_SET_POWER,
+    TRAMP_FRAME_SET_ACTIVE_STATE,
+    TRAMP_FRAME_GET_CONFIG,
+} trampFrameType_t;
+
+uint8_t trampCrc(trampFrame_t *frame);
+bool trampFrameInit(trampFrameType_t frameType, trampFrame_t *frame);
+void trampFrameClose(trampFrame_t *frame);
+
+uint8_t trampCrc(trampFrame_t *frame)
+{
+    uint8_t crc = 0;
+    uint8_t *ptr = (uint8_t *)frame;
+    for (int i = 1; i < TRAMP_HEADER_LENGTH + TRAMP_PAYLOAD_LENGTH; i++) {
+        crc += ptr[i];
+    }
+    return crc;
+}
+
+bool trampFrameInit(trampFrameType_t frameType, trampFrame_t *frame)
+{
+    frame->header.syncStart = TRAMP_SYNC_START;
+    switch (frameType) {
```

Comment:
While I always appreciate additional validation checks... :)  Is it necessary to have this switch  statement block and couldn't you just do
```
            frame->header.command = frameType;
```
`trampFrameInit()` is not a public function anyway and you have control over all the places it's called so that should implicitly provide the validation.

Reply:
I second that

Reply:
Or array may be used, it mapping from enum is desired ..

Reply:
Makes sense.. I made this more private as I went along and didn't clean up the mess.  I'll simplify this.

Reply:
Done